go/integTest/glide_test_suite_test.go

```diff
@@ -275,6 +286,29 @@ func (suite *GlideTestSuite) runWithClients(clients []api.BaseClient, test func(
 	}
 }
 
+func (suite *GlideTestSuite) runParallelizedWithClients(
+	clients []api.BaseClient,
+	parallelism int,
+	count int64,
+	test func(client api.BaseClient),
+) {
+	for _, client := range clients {
```

Comment:
Can you add there a timeout to kill or abandon all outstanding tasks once timed out?
In case if a test function or complete WaitGroup hang, it won't block other tests or CI.

Reply:
The timeout has been added too. Note that it is possible the failure message caused by the timeout is not related to timeout directly but something like "signal arrived during cgo execution".